lib/js/src/manager/screen/utils/_PresentAlertOperation.js

```diff
@@ -0,0 +1,501 @@
+        if (this._alertView.getAudio() === null || this._alertView.getAudio() === undefined) {
+            console.log('No audio sent for alert');
+            return true;
+        }
+        if (!this.supportsAlertAudioFile()) {
+            console.log('Module does not support audio files for alerts, skipping upload of audio files');
+            return true;
+        }
+        if (this._alertView.getAudio()._getAudioFiles() === null || this._alertView.getAudio()._getAudioFiles() === undefined || this._alertView.getAudio()._getAudioFiles().size === 0) {
+            console.log('No audio files to upload for alert');
+            return true;
+        }
+
+        const filesToBeUploaded = [];
+        for (const ttsChunk of this._alertView.getAudio().getAudioData()) {
+            if (ttsChunk.getType() !== SpeechCapabilities.FILE) {
+                continue;
+            }
+
+            const audioFile = this._alertView.getAudio()._getAudioFiles().get(ttsChunk.getText());
+            if (this._fileManager === null || !this._fileManager.fileNeedsUpload(audioFile)) {
+                continue;
+            }
+
+            filesToBeUploaded.push(audioFile);
+        }
+
+        if (filesToBeUploaded.length === 0) {
+            console.log('No audio files need to be uploaded for alert');
+            return true;
+        }
+
+        console.log('Uploading audio files for alert');
+
+        if (this._fileManager !== null || this._fileManager !== undefined) {
+            const successes = await this._fileManager.uploadFiles(filesToBeUploaded);
+
+            if (this.getState() === _Task.CANCELED) {
+                this._finishOperation(false, null);
+                return false;
+            }
+            if (successes !== null && successes !== undefined && successes.includes(false)) {
+                console.log('Error uploading alert audio files');
+            } else {
+                console.log('All alert audio files uploaded');
+            }
+            return true;
+        }
+    }
+
+    /**
+     * Checks whether the _AlertView has images that need to be uploaded, then uploads them.
+     * @returns {Promise} - resoves to Boolean.
+     */
+    async uploadImages () {
+        const artworksToBeUploaded = [];
+
+        if (this.supportsAlertIcon() && this._fileManager !== null && this._fileManager.fileNeedsUpload(this._alertView.getIcon())) {
+            artworksToBeUploaded.push(this._alertView.getIcon());
+        }
+
+        if (this._alertView.getSoftButtons() !== null && this._alertView.getSoftButtons() !== undefined) {
+            for (const object of this._alertView.getSoftButtons()) {
+                if (this.supportsSoftButtonImages() && object.getCurrentState() !== null && object.getCurrentState() !== undefined && this._fileManager !== null && this._fileManager.fileNeedsUpload(object.getCurrentState().getArtwork())) {
+                    artworksToBeUploaded.push(object.getCurrentState().getArtwork());
```

Comment:
we are limiting SoftButtons that will be set to 4, So we changed logic dealing with SoftButtons